book/chapter4/nonlinear_response/NLTHA_SDOF.tex

```diff
@@ -55,15 +62,94 @@
+
+\subsection{Multiple Stripe Analysis}
+\label{subsec:NLTHA-MSA}
+The Multiple Stripe Analysis (MSA) consists of applying a set of ground motion records that are scaled to multiple levels of intensity measure (intensity measure bins). Multiple “stripes” of structural response are thus obtained from the SDOF oscillator subjected to the ground motion records, as depicted in Figure \ref{fig:msa}.\
+
+\begin{figure}[htb]
+  \centering
+      \includegraphics[width=9cm]{figures/MSA_example.jpg}
+  \caption{Multiple stripes of structural responses obtained from MSA.}
+  \label{fig:msa}
+\end{figure}
+
+The response of the SDOF system to each ground motion record is used to determine the Probability Damage Matrix (PDM). In this case the PDM represents the number of records leading the structure to each damage state for the intensity measure of each "stripe" of responses. With MSA it is possible to derive fragility curves also for a single structure. Alternatively more capacity curves can be input and the PDMs of the corresponding SDOF systems are summed up to get a unique PDM for the building class.
+
+In order to to run the Multiple Stripe Analysis the User should specify the number of intensity measure bins and the number of records per bin, using the following variables:
+
+\begin{Verbatim}[frame=single, commandchars=\\\{\}, samepage=true]
+no_bins = 10
+no_rec_bin = 30
+\end{Verbatim}
+
+The User should also specify the scaling factors to apply to the set of ground motion records. Another set of files is thus necessary: for each of the Intensity Measure (IM) bin a csv file must be input, containing the name of the records and the corresponding scaling factor to scale them to the given IM level. The number of records in each file should be at least equal to the number of records per intensity measure bin defined in the variable \verb=no_rec_bin=. An example of the csv file for a single intensity measure bin is given below:
+
+\begin {table}[htb]
+\caption{Example of file containing scaling factor for each ground motion record}
+\label{table:pinching4}
+\begin{center}
+  \begin{tabular}{ | c | c |}
+  \hline
+	IN0031xa.csv & 1.008\\ \hline
+	IN0416xa.csv & 1.02\\ \hline
+	IN0192xa.csv & 0.977\\ \hline
+	IN0089xa.csv & 0.969\\ \hline
+	IN0344xa.csv & 0.962\\ \hline
+	... & ... \\ \hline
+	\end{tabular}
+\end{center}
+\end{table}
+
+The path to the folder where the csv files for each intensity measure bin are should be defined in the variable \verb=record_scaled_folder= (only the csv files containing the scaling factors should be in the folder) and the path to the folder where the ground motion records are should be specified in the variable \verb=gmrs_folder=, as exemplified below. The ground motion records are loaded as explained in Section \ref{subsec:gmrs}.
+
+\begin{Verbatim}[frame=single, commandchars=\\\{\}, samepage=true]
+gmrs_folder = "../../../../../rmtk_data/MSA_records"
+record_scaled_folder = "../../../../../rmtk_data/Scaling_factors"
```

Comment:
Yes it does. The files contain the name of the records and the corresponding scaling factor, so that the the IM of each record can be computed. The dynamic analyses are run with the order given in each file, so that the results are "clustered" for IM bin, in the sense that even if the IM bins are not ordered, for sure the results of the same IM bin are all together. Given that the number of bins and the number of records per bin are specified from the user, the script computed the IM of the first record of each "cluster" (which is the same of the rest of the records) and then it sorts the IMLs and the corresponding results.